commit a999b43ebf6300f2a09532527d49dc1074297d69 (refs/changes/67/4867/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-09-24T20:18:13-07:00 (1 year ago)
    
    TRITON-731 sdc-docker crashes on an active docker VM missing internal_metadata['docker:id']
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/backends/sdc/containers.js b/lib/backends/sdc/containers.js
index c4ed0ba..6e7be45 100644
--- a/lib/backends/sdc/containers.js
+++ b/lib/backends/sdc/containers.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var child_process = require('child_process');
@@ -2636,7 +2636,10 @@ function getContainers(opts, callback) {
                     // *strings*, so find any containers matching
                     // the requested values.
                     objects = objects.filter(function _filterId(entry) {
-                        var id = entry.internal_metadata['docker:id'];
+                        var id = utils.dockerIdFromVmObj(entry);
+                        if (!id) {
+                            return false;
+                        }
                         for (var i = 0; i < val.length; i++) {
                             if (id.match(val[i])) {
                                 return true;
diff --git a/package.json b/package.json
index 65e19a9..040d4bc 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "sdc-docker",
-  "version": "0.5.2",
+  "version": "0.5.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
